scc/src/opts.rs

```diff
@@ -50,85 +50,189 @@ pub fn fix_args(args: Vec<String>) -> Vec<String> {
-            };
-            if cur_name != name {
-                Err(format!("{} is not a known toggle option name", cur_name))
+            // println!("no_space: {}", s);
+            s.strip_prefix(name.to_owned().as_str())
+                .ok_or(Err::<String, &str>("IDK1"))
+                .map_or(Err("IDK2"), |s| Ok(s.to_string()))
+        })
+        .anywhere()
+        .optional()
+        .catch()
+}
+
+fn toggle_options(name: &'static str, help: &'static str) -> impl Parser<Option<bool>> {
+    any::<String>(name)
+        .help(help)
+        .guard(|s| s.eq(name.to_owned().as_str()), "toggle_option isn't -<name>")
+        .parse(move |s| {
+            // println!("toggle_options: {}", s);
+            if s != name {
+                Err(format!("{} is not a known toggle option name", s))
             } else {
                 Ok(true)
             }
         })
+        .adjacent()
         .anywhere()
-        .fallback(false)
+        .optional()
+        .catch()
 }
 
-fn slong(tag_str: &'static str, arg_str: &'static str) -> impl Parser<String> {
-    let tag = any::<String>(tag_str).guard(move |s| s.as_str() == tag_str, "not arg name");
-    let value = positional::<String>(arg_str);
-    construct!(tag, value).anywhere().map(|pair| pair.1)
+fn equals_sign(name: &'static str, help: &'static str) -> impl Parser<Option<String>> {
+    any::<String>(name)
+        .help(help)
+        .guard(
+            |s| s.starts_with(format!("{}=", name.to_owned()).as_str()),
+            "equals_sign doesn't start with -<name>=",
+        )
+        .parse(move |s| {
+            // println!("equals_sign: {}", s);
+            let arg = s.split('=').collect::<Vec<_>>();
+            let cur_name = arg.first().unwrap().to_owned();
+            if cur_name != name {
+                Err(format!("{} is not a known option name", cur_name))
+            } else {
+                Ok((*arg.last().unwrap()).to_string())
+            }
+        })
+        .anywhere()
+        .optional()
+        .catch()
+}
+
+fn slong(name: &'static str, arg_name: &'static str, help: &'static str) -> impl Parser<Option<String>> {
+    let tag = any::<String>(name)
+        .help(help)
+        .guard(|s| s.eq(name.to_owned().as_str()), "slong isn't -<name>");
+    let value = positional::<String>(arg_name).guard(|s| !s.starts_with('-'), "starts with -");
+    construct!(tag, value)
+        .adjacent()
+        .anywhere()
+        .map(|pair| pair.1)
+        .optional()
+        .catch()
 }
 
 fn script_paths() -> impl Parser<Vec<PathBuf>> {
     let tag = any::<String>("-compile").guard(|s| s == "-compile", "not compile");
-    let value = positional::<PathBuf>("SCRIPT_PATH");
-    construct!(tag, value).anywhere().map(|pair| pair.1).many().catch()
+    let value = positional::<PathBuf>("SCRIPT_PATH")
+        .guard(|s| !s.to_str().unwrap().to_string().starts_with('-'), "starts with -");
```

Comment:
this also should not use `unwrap`
```suggestion
        .guard(|s| matches!(s.to_str(), Some(str) if !str.starts_with('-')), "starts with -");
```